strawberry/schema/schema.py

```diff
@@ -334,7 +334,16 @@ def _resolve_node_ids(self):
             # early feedback for missing NodeID annotations
             origin = type_def.origin
             if issubclass(origin, relay.Node):
-                origin.resolve_id_attr()
+                has_custom_resolve_id = False
+                for base in origin.__mro__:
+                    if base is relay.Node:
+                        break
+                    if "resolve_id" in base.__dict__:
+                        has_custom_resolve_id = True
+                        break
+
+                if not has_custom_resolve_id:
+                    origin.resolve_id_attr()
```

Comment:
could this be baked into Node itself?

Reply:
maybe with __init_subclass__

Reply:
can you please test the change? Currently because of pydantic errors I cannot test

Reply:
this would prevent non-final Node subclasses.

Reply:
I may be wrong, but I imagine what @patrick91 suggested is to move the logic to move this logic to inside `resolve_id_attr` itself

Reply:
this breaks the guarantee that `resolve_id_attr` always returns an attribute or error.

It would be better to have an hook on Node and possible other types: `add_to_schema(cls, schema)`. This would allow a type agnostic customization

Reply:
but I think this should be scope of a different PR.